fs/operations/operations.go

```diff
@@ -361,10 +361,21 @@ func Copy(ctx context.Context, f fs.Fs, dst fs.Object, remote string, src fs.Obj
 	if !ci.Inplace && f.Features().Move != nil && f.Features().PartialUploads && !strings.HasSuffix(remote, ".rclonelink") {
 		// Avoid making the leaf name longer if it's already lengthy to avoid
 		// trouble with file name length limits.
-		suffix := "." + random.String(8) + ".partial"
+		ext := strings.TrimSpace(ci.PartialUploadExtension)
+		if len(ext) == 0 {
+			ext = fs.GetConfig(context.Background()).PartialUploadExtension
+		}
+
+		suffix := "." + random.String(8) + "." + ext
 		base := path.Base(remotePartial)
 		if len(base) > 100 {
-			remotePartial = remotePartial[:len(remotePartial)-len(suffix)] + suffix
+			limit := len(base) - len(suffix)
```

Comment:
I think since we've limited the size of the extension, we can just use the original 

    remotePartial = remotePartial[:len(remotePartial)-len(suffix)] + suffix

Here, which will give a max total length of 125 which is fine.